### PR TITLE
WINC-649: [wmco] Change BYOH annotation to a label

### PR DESCRIPTION
**Pull request by sebsoto**
Changes the BYOH annotation to a label to allow for easier selection of
BYOH nodes.

### CODE REVIEW

**Review by saifshaikh48 — APPROVED** [review]
Thanks for working on this --

/lgtm

**Comment by mansikulkarni96** [comment]
LGTM

File: controllers/configmap_controller.go [hunk 1]
```diff
@@ -219,7 +219,7 @@ func (r *ConfigMapReconciler) ensureInstancesAreUpToDate(instances []*instances.
 func (r *ConfigMapReconciler) deconfigureInstances(instances []*instances.InstanceInfo, nodes *core.NodeList) error {
 	for _, node := range nodes.Items {
 		// Only looking at BYOH nodes
-		if _, present := node.Annotations[BYOHAnnotation]; !present {
+		if _, present := node.GetLabels()[BYOHLabel]; !present {
```
Comment (aravindhp) [inline controllers/configmap_controller.go:222]: I suggest updating the filter when you are getting the list of nodes.

File: controllers/secret_controller.go [hunk 2]
```diff
@@ -177,7 +177,7 @@ func (r *SecretReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 		for _, node := range nodes.Items {
 			// Only clear the annotation for Nodes associated with Machines, as the clearing of the annotation is used
 			// solely to kick off the deletion and recreation of Machines.
-			if _, present := node.Annotations[BYOHAnnotation]; present {
+			if _, present := node.GetLabels()[BYOHLabel]; present {
```
Comment (aravindhp) [inline controllers/secret_controller.go:180]: I suggest updating the filter when you are getting the list of nodes.

**Review by aravindhp — CHANGES_REQUESTED** [review]
Thanks for doing this, @sebsoto. Please address my comment.

File: controllers/configmap_controller.go (hunk 1 above)
Comment (sebsoto) [reply controllers/configmap_controller.go]: Done.

File: controllers/secret_controller.go (hunk 2 above)
Comment (sebsoto) [reply controllers/secret_controller.go:180]: I did not want to do this as we need both BYOH and Machine nodes for https://github.com/openshift/windows-machine-config-operator/pull/508

File: controllers/configmap_controller.go [hunk 3]
```diff
@@ -169,8 +169,9 @@ func validateAddress(address string) error {
 
 // reconcileNodes corrects the discrepancy between the "expected" instances, and the "actual" Node list
 func (r *ConfigMapReconciler) reconcileNodes(ctx context.Context, windowsInstances *core.ConfigMap) error {
+	// Get the current list of Windows BYOH Nodes
 	nodes := &core.NodeList{}
-	if err := r.client.List(ctx, nodes); err != nil {
+	if err := r.client.List(ctx, nodes, client.MatchingLabels{BYOHLabel: "true"}); err != nil {
```
Comment (aravindhp) [inline controllers/configmap_controller.go:174]: To be on the safe side, the Windows label should also be added.

File: controllers/configmap_controller.go [hunk 4]
```diff
@@ -218,10 +219,6 @@ func (r *ConfigMapReconciler) ensureInstancesAreUpToDate(instances []*instances.
 // deconfigures the instances associated with them.
 func (r *ConfigMapReconciler) deconfigureInstances(instances []*instances.InstanceInfo, nodes *core.NodeList) error {
 	for _, node := range nodes.Items {
-		// Only looking at BYOH nodes
-		if _, present := node.Annotations[BYOHAnnotation]; !present {
-			continue
-		}
```
Comment (aravindhp) [inline controllers/configmap_controller.go:224]: This implies the method doc needs to be updated to state that that input nodes are assumed to be Windows BYOH

**Review by aravindhp — CHANGES_REQUESTED** [review]

**Comment by mansikulkarni96** [comment]
Thanks for working on this @sebsoto 
/lgtm

**Review by jrvaldes — APPROVED** [review]
/lgtm

**Review by aravindhp — APPROVED** [review]
Thanks for doing this, @sebsoto 

/approve

**Comment by sebsoto** [comment]
Machine provisioning flake, not an issue on our side
/retest

**Pull request merged**